src/Appwrite/Platform/Tasks/ScheduleExecutions.php

```diff
@@ -22,11 +21,16 @@ public static function getSupportedResource(): string
         return 'execution';
     }
 
-    protected function enqueueResources(Group $pools, Database $dbForConsole): void
+    protected function enqueueResources(array $pools, callable $getConsoleDB): void
     {
-        $queue = $pools->get('queue')->pop();
-        $connection = $queue->getResource();
-        $queueForFunctions = new Func($connection);
+        [$connection,$pool, $dbForConsole] = $getConsoleDB();
+        $this->connections->add($connection, $pool);
+
+        $pool = $pools['pools-queue-queue']['pool'];
+        $connection = $pool->get();
+        $this->connections->add($connection, $pool);
```

Comment:
why are we adding the connection twice?

Reply:
The first one is for the `consoleDB`, and the second is for the `queue.`
I'll rename it for better clarity